### ISSUE TITLE
What Node version is needed? (ERR_REQUIRE_ESM require() error)

### ISSUE THREAD

**Issue by ipkpjersi** [peterthehan/runescape-text#30]
Hi,

I tried getting this to work but I've not had much luck with Node 16 or 18:
```
➜ user@good-computer  ~  npm i runescape-text
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'runescape-text@3.1.0',
npm WARN EBADENGINE   required: { node: '>=18' },
npm WARN EBADENGINE   current: { node: 'v16.18.1', npm: '8.19.2' }
npm WARN EBADENGINE }
npm WARN deprecated inflight@1.0.6: This module is not supported, and leaks memory. Do not use it. Check out lru-cache if you want a good and tested way to coalesce async requests by a key value, which is much more comprehensive and powerful.
npm WARN deprecated npmlog@5.0.1: This package is no longer supported.
npm WARN deprecated rimraf@3.0.2: Rimraf versions prior to v4 are no longer supported
npm WARN deprecated are-we-there-yet@2.0.0: This package is no longer supported.
npm WARN deprecated glob@7.2.3: Glob versions prior to v9 are no longer supported
npm WARN deprecated gauge@3.0.2: This package is no longer supported.

added 66 packages, and audited 68 packages in 4s

7 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
npm notice 
npm notice New major version of npm available! 8.19.2 -> 10.8.3
npm notice Changelog: https://github.com/npm/cli/releases/tag/v10.8.3
npm notice Run npm install -g npm@10.8.3 to update!
npm notice 
➜ user@good-computer  ~  npx runescape-text "glow3:wave:hello world"
/home/user/node_modules/runescape-text/dist/classes/Encoder.js:3
const gifenc_1 = require("gifenc");
                 ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /home/user/node_modules/gifenc/src/index.js from /home/user/node_modules/runescape-text/dist/classes/Encoder.js not supported.
Instead change the require of index.js in /home/user/node_modules/runescape-text/dist/classes/Encoder.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/home/user/node_modules/runescape-text/dist/classes/Encoder.js:3:18)
    at Object.<anonymous> (/home/user/node_modules/runescape-text/dist/index.js:9:35)
    at Object.<anonymous> (/home/user/node_modules/runescape-text/dist/cli.js:17:33) {
  code: 'ERR_REQUIRE_ESM'
}
➜ user@good-computer  ~  
➜ user@good-computer  ~  npm install gifenc
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'runescape-text@3.1.0',
npm WARN EBADENGINE   required: { node: '>=18' },
npm WARN EBADENGINE   current: { node: 'v16.18.1', npm: '8.19.2' }
npm WARN EBADENGINE }

up to date, audited 68 packages in 2s

7 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
➜ user@good-computer  ~  npx runescape-text "glow3:wave:hello world"
/home/user/node_modules/runescape-text/dist/classes/Encoder.js:3
const gifenc_1 = require("gifenc");
                 ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /home/user/node_modules/gifenc/src/index.js from /home/user/node_modules/runescape-text/dist/classes/Encoder.js not supported.
Instead change the require of index.js in /home/user/node_modules/runescape-text/dist/classes/Encoder.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/home/user/node_modules/runescape-text/dist/classes/Encoder.js:3:18)
    at Object.<anonymous> (/home/user/node_modules/runescape-text/dist/index.js:9:35)
    at Object.<anonymous> (/home/user/node_modules/runescape-text/dist/cli.js:17:33) {
  code: 'ERR_REQUIRE_ESM'
}
➜ user@good-computer  ~  nvm install 18; nvm use 18; npx runescape-text "glow3:wave:hello world"
Downloading and installing node v18.20.4...
Downloading https://nodejs.org/dist/v18.20.4/node-v18.20.4-linux-x64.tar.xz...
################################################################################### 100.0%
Computing checksum with sha256sum
Checksums matched!
Now using node v18.20.4 (npm v10.7.0)
Now using node v18.20.4 (npm v10.7.0)
node:internal/modules/cjs/loader:1460
  return process.dlopen(module, path.toNamespacedPath(filename));
                 ^

Error: The module '/home/user/node_modules/canvas/build/Release/canvas.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 93. This version of Node.js requires
NODE_MODULE_VERSION 108. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
    at Module._extensions..node (node:internal/modules/cjs/loader:1460:18)
    at Module.load (node:internal/modules/cjs/loader:1203:32)
    at Module._load (node:internal/modules/cjs/loader:1019:12)
    at Module.require (node:internal/modules/cjs/loader:1231:19)
    at require (node:internal/modules/helpers:177:18)
    at Object.<anonymous> (/home/user/node_modules/canvas/lib/bindings.js:3:18)
    at Module._compile (node:internal/modules/cjs/loader:1364:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1422:10)
    at Module.load (node:internal/modules/cjs/loader:1203:32)
    at Module._load (node:internal/modules/cjs/loader:1019:12) {
  code: 'ERR_DLOPEN_FAILED'
}

Node.js v18.20.4
➜ user@good-computer  ~  npm install runescape-text

up to date, audited 68 packages in 710ms

7 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
➜ user@good-computer  ~  npx runescape-text "glow3:wave:hello world" 
node:internal/modules/cjs/loader:1460
  return process.dlopen(module, path.toNamespacedPath(filename));
                 ^

Error: The module '/home/user/node_modules/canvas/build/Release/canvas.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 93. This version of Node.js requires
NODE_MODULE_VERSION 108. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
    at Module._extensions..node (node:internal/modules/cjs/loader:1460:18)
    at Module.load (node:internal/modules/cjs/loader:1203:32)
    at Module._load (node:internal/modules/cjs/loader:1019:12)
    at Module.require (node:internal/modules/cjs/loader:1231:19)
    at require (node:internal/modules/helpers:177:18)
    at Object.<anonymous> (/home/user/node_modules/canvas/lib/bindings.js:3:18)
    at Module._compile (node:internal/modules/cjs/loader:1364:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1422:10)
    at Module.load (node:internal/modules/cjs/loader:1203:32)
    at Module._load (node:internal/modules/cjs/loader:1019:12) {
  code: 'ERR_DLOPEN_FAILED'
}

Node.js v18.20.4
➜ user@good-computer  ~  npm rebuild runescape-text          
rebuilt dependencies successfully
➜ user@good-computer  ~  npx runescape-text "glow3:wave:hello world"
node:internal/modules/cjs/loader:1460
  return process.dlopen(module, path.toNamespacedPath(filename));
                 ^

Error: The module '/home/user/node_modules/canvas/build/Release/canvas.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 93. This version of Node.js requires
NODE_MODULE_VERSION 108. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
    at Module._extensions..node (node:internal/modules/cjs/loader:1460:18)
    at Module.load (node:internal/modules/cjs/loader:1203:32)
    at Module._load (node:internal/modules/cjs/loader:1019:12)
    at Module.require (node:internal/modules/cjs/loader:1231:19)
    at require (node:internal/modules/helpers:177:18)
    at Object.<anonymous> (/home/user/node_modules/canvas/lib/bindings.js:3:18)
    at Module._compile (node:internal/modules/cjs/loader:1364:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1422:10)
    at Module.load (node:internal/modules/cjs/loader:1203:32)
    at Module._load (node:internal/modules/cjs/loader:1019:12) {
  code: 'ERR_DLOPEN_FAILED'
}

Node.js v18.20.4
➜ user@good-computer  ~  npm rebuild canvas
rebuilt dependencies successfully
➜ user@good-computer  ~  npx runescape-text "glow3:wave:hello world"
/home/user/node_modules/runescape-text/dist/classes/Encoder.js:3
const gifenc_1 = require("gifenc");
                 ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /home/user/node_modules/gifenc/src/index.js from /home/user/node_modules/runescape-text/dist/classes/Encoder.js not supported.
Instead change the require of index.js in /home/user/node_modules/runescape-text/dist/classes/Encoder.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/home/user/node_modules/runescape-text/dist/classes/Encoder.js:3:18)
    at Object.<anonymous> (/home/user/node_modules/runescape-text/dist/index.js:9:35)
    at Object.<anonymous> (/home/user/node_modules/runescape-text/dist/cli.js:17:33) {
  code: 'ERR_REQUIRE_ESM'
}

Node.js v18.20.4
➜ user@good-computer  ~  npx runescape-text "glow3:wave:hello world"
/home/user/node_modules/runescape-text/dist/classes/Encoder.js:3
const gifenc_1 = require("gifenc");
                 ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /home/user/node_modules/gifenc/src/index.js from /home/user/node_modules/runescape-text/dist/classes/Encoder.js not supported.
Instead change the require of index.js in /home/user/node_modules/runescape-text/dist/classes/Encoder.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/home/user/node_modules/runescape-text/dist/classes/Encoder.js:3:18)
    at Object.<anonymous> (/home/user/node_modules/runescape-text/dist/index.js:9:35)
    at Object.<anonymous> (/home/user/node_modules/runescape-text/dist/cli.js:17:33) {
  code: 'ERR_REQUIRE_ESM'
}

Node.js v18.20.4
```

What node version is needed for this to work?

Cheers.

**Comment by peterthehan** [comment]
oh hmm, could you try the LTS version of node? on a separate note, i guess i should bump the deps since it's been a while... 😂

**Comment by ipkpjersi** [comment]
Node 16 was LTS IIRC, but I just tried Node 20 LTS and had to do `sudo apt-get install build-essential libcairo2-dev libjpeg-dev libpango1.0-dev libgif-dev librsvg2-dev` to get canvas to re-build with Node 20, but still same require() error.

**Comment by peterthehan** [comment]
@ipkpjersi should be fixed with v3.1.1 🤞 lmk if there are still issues

**Comment by ipkpjersi** [comment]
Yeah it works now, thanks :)
